e2e/tests/truffle/test/Event.js

```diff
@@ -0,0 +1,240 @@
+const Web3 = require('web3')
+const fs = require('fs')
+const path = require('path')
+
+const {
+    SpeculativeNonceTxMiddleware,
+    SignedTxMiddleware,
+    Client,
+    LocalAddress,
+    CryptoUtils,
+    LoomProvider
+} = require('loom-js')
+
+const EventTestContract = artifacts.require('EventTestContract')
+
+contract('EventTestContract', async (accounts) => {
+    let contract, from, nodeAddr, contractAddress, web3eth
+    const newValueSetEventTopic = "0xb922f092a64f1a076de6f21e4d7c6400b6e55791cc935e7bb8e7e90f7652f15b"
+    const anotherValueSetEventTopic = "0xccdec039614c54f7047ec6e5fbd3b6b8d5d78e1a38737dcc140cf162a774a83f"
+    beforeEach(async () => {
+        nodeAddr = fs.readFileSync(path.join(process.env.CLUSTER_DIR, '0', 'node_rpc_addr'), 'utf-8').trim()
+        const chainID = 'default'
+        const writeUrl = `ws://${nodeAddr}/websocket`
+        const readUrl = `ws://${nodeAddr}/queryws`
+
+        const privateKey = CryptoUtils.generatePrivateKey()
+        const publicKey = CryptoUtils.publicKeyFromPrivateKey(privateKey)
+
+        from = LocalAddress.fromPublicKey(publicKey).toString()
+
+        var client = new Client(chainID, writeUrl, readUrl)
+        client.on('error', msg => {
+            console.error('Error on connect to client', msg)
+            console.warn('Please verify if loom cluster is running')
+        })
+        const setupMiddlewareFn = function (client, privateKey) {
+            const publicKey = CryptoUtils.publicKeyFromPrivateKey(privateKey)
+            return [new SpeculativeNonceTxMiddleware(publicKey, client), new SignedTxMiddleware(privateKey)]
+        }
+        var loomProvider = new LoomProvider(client, privateKey, setupMiddlewareFn)
+
+        let web3 = new Web3(loomProvider)
+        let eventTestContract = await EventTestContract.deployed()
+        contractAddress = eventTestContract.address
+        contract = new web3.eth.Contract(EventTestContract._json.abi, contractAddress, {
+            from
+        });
+        web3eth = new Web3(new Web3.providers.WebsocketProvider(`ws://${nodeAddr}/eth`));
+    })
+
+    it('contract.events.allEvents() & web3.eth.subscribe()', async () => {
+        try {
+            var eventCount = 0
+            // Test loom
+            contract.events.allEvents()
+                .on('data', (event) => {
+                    eventCount++
+                })
+            // Test /eth endpoint
+            var ethEventCount = 0
+            web3eth.eth.subscribe('logs', {
+                address: contractAddress,
+            }, function (error, result) {
+                if (!error) {
+                    ethEventCount++
+                }
+            });
+            var tx = await contract.methods.set(1).send()
+            assert.equal(2, tx.events.NewValueSet.length)
+            assert.equal(2, tx.events.AnotherValueSet.logIndex)
```

Comment:
Can you also check the values in the events match the expected numbers?

Reply:
Done